core/templates/domain/objects/UnitsObjectFactory.ts

```
@@ -114,7 +118,11 @@ export class UnitsObjectFactory {
         }
       } else if (unitList[ind] === ')') {
         var elem = parenthesisStack.pop();
-        multiplier = parseInt(elem[1]) * multiplier;
+        // If the parenthesisStack is empty then we don't need to
+        // modify the value of multiplier and this is being done
+        // using multiplicative identity property.
+        var parsed: number = elem ? parseInt(<string> elem[1]) : 1;
+        multiplier = parsed * multiplier;
```

Now thinking about this can it actually happen that `elem` is `null` or `undefined`? If not we should maybe rather raise an error if it happens.

---

yes, logically it should not happen.